### PR TITLE
Add unit tests for internal/testutil and internal/toolsets/mock

### PR DESCRIPTION
**Pull request by janisz**
### Description

Add comprehensive unit tests with high coverage:
- internal/testutil: 95.8% coverage (ports, server, config)
- internal/toolsets/mock: 100% coverage (tool, toolset)

Tests cover normal operation, edge cases, error handling, and interface compliance verification.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### Validation

CI

### CODE REVIEW

File: internal/testutil/config_test.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+	// The file should be in a temp directory managed by t.TempDir()
+	// which will be automatically cleaned up
+	dir := filepath.Dir(filePath)
+	if dir == "" || dir == "." {
+		t.Errorf("File should be in a proper directory, got: %s", dir)
+	}
+}
+
+func TestWriteYAMLFile_LargeContent(t *testing.T) {
```
Comment (mtodor) [inline internal/testutil/config_test.go]: Can be deleted.

File: internal/testutil/ports_test.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		if port < minPort || port >= maxPort {
+			t.Errorf("Port %d is outside valid range [%d, %d)", port, minPort, maxPort)
+		}
+	})
+
+	t.Run("returns deterministic port for same test name", func(t *testing.T) {
+		port1 := GetPortForTest(t)
+		port2 := GetPortForTest(t)
+
+		if port1 != port2 {
```
Comment (mtodor) [inline internal/testutil/ports_test.go]: Let's use assert.

File: internal/toolsets/mock/toolset_test.go [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+
+func TestToolset_GetName(t *testing.T) {
+	t.Run("returns configured name", func(t *testing.T) {
+		name := "my-toolset"
+		toolset := NewToolset(name, true, nil)
+
+		assert.Equal(t, name, toolset.GetName())
+	})
+
+	t.Run("returns empty string if configured", func(t *testing.T) {
```
Comment (mtodor) [inline internal/toolsets/mock/toolset_test.go]: Can be deleted.
